tools/integration_tests/util/operations/file_operations.go

```diff
@@ -315,51 +315,82 @@ func StatFile(file string) (*fs.FileInfo, error) {
 // If sizes match, then compares the contents of both the files.
 // Not a good idea for very large files as it loads both the files' contents in
 // the memory completely.
-// Returns 0 if no error and files match.
-// Returns 1 if files don't match and captures reason for mismatch in err.
-// Returns 2 if any error.
-func DiffFiles(filepath1, filepath2 string) (int, error) {
+// Returns true if no error and files match.
+// Returns false if files don't match (captures reason for mismatch in err) or if any other error.
+func AreFilesIdentical(filepath1, filepath2 string) (bool, error) {
 	if filepath1 == "" || filepath2 == "" {
-		return 2, fmt.Errorf("one or both files being diff'ed have empty path")
+		return false, fmt.Errorf("one or both files being diff'ed have empty path")
 	} else if filepath1 == filepath2 {
-		return 0, nil
+		return true, nil
 	}
 
 	fstat1, err := StatFile(filepath1)
 	if err != nil {
-		return 2, err
+		return false, err
 	}
 
 	fstat2, err := StatFile(filepath2)
 	if err != nil {
-		return 2, err
+		return false, err
 	}
 
 	file1size := (*fstat1).Size()
 	file2size := (*fstat2).Size()
 	if file1size != file2size {
-		return 1, fmt.Errorf("files don't match in size: %s (%d bytes), %s (%d bytes)", filepath1, file1size, filepath2, file2size)
+		return false, fmt.Errorf("files don't match in size: %s (%d bytes), %s (%d bytes)", filepath1, file1size, filepath2, file2size)
 	}
 
-	bytes1, err := ReadFile(filepath1)
-	if err != nil || bytes1 == nil {
-		return 2, fmt.Errorf("failed to read file %s", filepath1)
-	} else if int64(len(bytes1)) != file1size {
-		return 2, fmt.Errorf("failed to completely read file %s", filepath1)
+	if file1size == 0 {
+		return true, nil
 	}
 
-	bytes2, err := ReadFile(filepath2)
-	if err != nil || bytes2 == nil {
-		return 2, fmt.Errorf("failed to read file %s", filepath2)
-	} else if int64(len(bytes2)) != file2size {
-		return 2, fmt.Errorf("failed to completely read file %s", filepath2)
+	f1, err := os.OpenFile(filepath1, os.O_RDONLY|syscall.O_DIRECT, FilePermission_0400)
+	if err != nil {
+		return false, fmt.Errorf("failed to read file %s: %v", filepath1, err)
+	}
+
+	defer CloseFile(f1)
+
+	f2, err := os.OpenFile(filepath2, os.O_RDONLY|syscall.O_DIRECT, FilePermission_0400)
+	if err != nil {
+		return false, fmt.Errorf("failed to read file %s: %v", filepath2, err)
 	}
 
-	if !bytes.Equal(bytes1, bytes2) {
-		return 1, fmt.Errorf("files don't match in content: %s, %s", filepath1, filepath2)
+	defer CloseFile(f2)
+
+	sizeRemaining := int(file1size)
+	const ChunkSize = 16000
```

Comment:
If its a constant why can't it be outside this method. I haven't seen constants inside functions. Is it recommended?

Reply:
I haven't read/seen anything against constants defined inside functions. 
It can be defined outside the function, yes. I defined it inside the function to limit its scope, because that's the only place it's used. Anyway I've moved it at file-level now.